packages/world/src/systems/libraries/MoveLib.sol

```diff
@@ -118,11 +118,16 @@ library MoveLib {
       setPlayer(newPlayerCoords[i], playerEntityIds[i]);
     }
 
+    uint128 currentEnergy = Energy._getEnergy(playerEntityId);
     uint128 energyCost = (PLAYER_MOVE_ENERGY_COST * uint128(newBaseCoords.length - numFalls)) +
       (PLAYER_FALL_ENERGY_COST * numFalls);
-    uint128 currentEnergy = Energy._getEnergy(playerEntityId);
-    transferEnergyToPool(playerEntityId, playerCoord, energyCost > currentEnergy ? currentEnergy : energyCost);
-    // TODO: drop inventory items
+    energyCost = energyCost > currentEnergy ? currentEnergy : energyCost;
+
+    // TODO: figure out how to run gravity and only kill AFTER that
```

Comment:
so now if the player dies while falling what happens? does gravity keep running or the transaction ends?

Reply:
That's what we need t figure out, we don't want players to die in the middle of a fall right, so we should probably wait until they hit the ground

Reply:
oh no we shouldn't wait for them to hit the ground! they should die in the middle of a fall

otherwise imagine you're standing on a mountain/floating island, and jump off. You'd have to pass gas to run gravity alll the way to the ground. vs the fall damage parameter is basically a way we can tune the max gas you have to pay to die, cuz we know for sure gravity will stop running after X blocks.

Reply:
ah interesting, but then the inventory would stay floating in the air

Reply:
make sense, would it become cheaper if we do that thing where the client provides the whole path (including gravity) ?

Reply:
we'd still have to verify the path in the contracts right? so wouldnt it be more expensive cuz compute is cheaper than call data?

Reply:
yeah maybe, haven't checked the move system in detail, but nvm we can look into it later

Reply:
can also make this todo into a separate github issue of dealing with gravity. I think when we fix it we should resolve the two missing tests in Gravity.t.sol as well that check for death

Reply:
I now made it so we just return that gravity doesn't apply if the energy was consumed while falling, so it will just kill the player mid air, will add tests for that after merge